akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -129,32 +130,40 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
             stashState = stashState)
 
           // needs to accept Any since we also can get messages from the journal
-          // not part of the protocol
-          val onStopInterceptor = new BehaviorInterceptor[Any, Any] {
+          // not part of the user facing Command protocol
+          def interceptor: BehaviorInterceptor[Any, InternalProtocol] = new BehaviorInterceptor[Any, InternalProtocol] {
```

Comment:
for the record, I verified with println that this was created and used correctly when the actor was restarted (e.g. not duplicate interceptors in the stack)